nominal/core/dataset.py

```diff
@@ -31,18 +31,32 @@
 from nominal.core._utils import HasRid, update_dataclass
 from nominal.core.channel import Channel, _get_series_values_csv
 from nominal.exceptions import NominalIngestError, NominalIngestFailed, NominalIngestMultiError
-from nominal.ts import _MAX_TIMESTAMP, _MIN_TIMESTAMP, _AnyTimestampType, _to_typed_timestamp_type
+from nominal.ts import _MAX_TIMESTAMP, _MIN_TIMESTAMP, _AnyTimestampType, _SecondsNanos, _to_typed_timestamp_type
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class DatasetBounds:
+    start: datetime
+    end: datetime
```

Comment:
we should use `IntegralNanosecondsUTC` here instead as we do for `Run` and others; `datetime` loses precision